server/src/main/java/io/crate/expression/scalar/conditional/MultiCaseFunction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.conditional;
+
+import static io.crate.metadata.functions.TypeVariableConstraint.typeVariable;
+import static io.crate.types.TypeSignature.parseTypeSignature;
+
+import java.util.List;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.BoundSignature;
+import io.crate.metadata.functions.Signature;
+import io.crate.types.DataTypes;
+
+public class MultiCaseFunction extends Scalar<Object, Object> {
+
+    public static void register(ScalarFunctionModule module) {
+        // multi_case([boolean] conditions, [E] results) -> E
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.BOOLEAN_ARRAY.getTypeSignature(),
+                parseTypeSignature("array(E)"),
+                parseTypeSignature("E")
+            ).withTypeVariableConstraints(typeVariable("E")),
+            MultiCaseFunction::new
+        );
+        // multi_case([boolean] conditions, [E] results, E default) -> E
+        module.register(
+            Signature.scalar(
+                NAME,
+                DataTypes.BOOLEAN_ARRAY.getTypeSignature(),
+                parseTypeSignature("array(E)"),
+                parseTypeSignature("E"),
+                parseTypeSignature("E")
+            ).withTypeVariableConstraints(typeVariable("E")),
+            MultiCaseFunction::new
+        );
+    }
+
+    public static final String NAME = "multi_case";
```

Comment:
maybe move this at the top of the class?

Reply:
Most of the other scalar functions also have the `NAME` after the register method. It's ok to move it up, no strong opinion here.

Reply:
also fine, It's more of personal preference.